xonsh/jobs.py

```diff
@@ -6,13 +6,29 @@
 import signal
 import builtins
 from subprocess import TimeoutExpired, check_output
-from io import BytesIO
 from collections import deque
 
-from xonsh.tools import ON_WINDOWS
+from xonsh.platform import ON_DARWIN, ON_WINDOWS
 
 tasks = deque()
 
+
+if ON_DARWIN:
+    def _send_signal(job, signal):
+        # On OS X, os.killpg() may cause PermissionError when there are
+        # any zombie processes in the process group.
+        # See github issue #1012 for details
+        for pid in job['pids']:
+            os.kill(pid, signal)
+
+elif ON_WINDOWS:
+    pass
```

Comment:
Why we need it on Windows? This is only a private helper function for Unix I guess.